src/cargo/sources/git/utils.rs

```diff
@@ -958,6 +964,8 @@ pub fn fetch(
 
     clean_repo_temp_files(repo);
 
+    let mut revision: String = String::new();
```

Comment:
Probably don't need this extra variable. `oid_to_fetch` already contains the revision information we want.